content/docs/developers/_index.md

```diff
@@ -0,0 +1,145 @@
+similar to Slack, Discord or IRC channels. Most of the events in such rooms are
+about sending messages. Rooms have a unique technical identifier, and zero or
+several human-readable aliases or addresses. Aliases are made of a room name,
+and a server part. A typical room alias would be:
+
+```
+#mountain-bike:example.com
+```
+
+Homeservers are federated: the Matrix specification defines a [Sever-Server API](https://spec.matrix.org/latest/server-server-api/)
+(also known as Federation API) to describe interactions between servers.
+Whenever a user is in a room, their homeserver needs to have a local copy of
+that room.
+
+For example, if `@alice:outdoors.com` is the first user from `outdoors.com` to
+try to join `#mountain-bike:example.com`, then her homeserver is going to reach
+out to `example.com` to get a copy of the room. `outdoors.com` and `example.com`
+then stay in touch to synchronise their copy of the room.
+
+Whenever the homeserver receives new events, it's in charge of parsing them,
+perform checks on the event, and take action accordingly (e.g. add a specific 
+user to a specific room when it receives a room join event). The expected
+behaviour of homeservers is described fully in the [Matrix Specification](https://spec.matrix.org).
+
+You can find a list of existing homeserver implementations in the
+[Ecosystem > Servers](/ecosystem/servers) section of this website. Some of them
+are open source, so you can explore how they work. Please also refer to their
+documentation if you want to deploy them either for testing or in production.
+
+### Client
+
+Homeservers communicate between each other with the Server-Server / Federation
+API, but they also communicate with clients in a standard way: the Client-Server
+API.
+
+Clients are pieces of software who can use a Matrix account to send and receive
+events from a specific homeserver. The clients themselves only ever talk to the
+homeserver of the account they're using. If a client uses the
+`@alice:example.com` account, they will only talk to `example.com`.
+
+The most common kinds of client are the user facing ones. In the case of instant
+messaging, those clients show rooms as timeline of messages, with users joining,
+leaving, redacting messages…
+
+To get a better idea of what clients look like in practice, you can find a list
+on [Ecosystem > Clients](/ecosystem/clients) and give them a go.
+
+If you're more interested in writing your own client to bring a new experience
+to users, you may want to rely on an existing SDK (see
+[Ecosystem > SDKs](/ecosystem/sdks)). Those will do significant part of the
+Matrix heavy-lifting and allow you to focus on the UX you want to build.
+Finally, if you're interested in learning more about the interactions between
+clients and servers, please head to
+[the Client-Server section of the Matrix Specification](https://spec.matrix.org/latest/client-server-api/).
+
+### AppService (bots and bridges)
```

Comment:
It's not actually that common for bots to be appservices, so this heading seems a bit misleading.

Reply:
I would kinda disagree. While in the public that is true there are some non public projects using appservices to be able to do bots at scale. One public project may be mjolnir which seems to be looking into using appservices to provide itself to multiple users.

So I think it makes sense to list both there. However it might make sense to say that appservices for bots only make sense of you have to deal with a scale where using /sync or a single bot user may not work anymore.
As in: appservices for bots are mostly a thing you need for scaling and should not be the default for bots.

Reply:
I'm just saying that bots are not a kind of appservice, which the title reads like to me. I'm totally in favor of having client-like and appservice-like bots both documented.

Reply:
I think it's worth keeping `(bots and bridges)` in the title because `appservice` is a very matrixy term. Newcomers will not necessarily know what appservices are. The first paragraph also says:

> Many Matrix bots are non-human clients. They can be built with the same SDKs as regular clients, and instead of showing a UI to display what is happening they will listen to events, parse them, and for example send automatic replies.

I'm wondering what's the best way to make it more obvious that regular clients can be bots too. Should I add (clients and bots) in the `### Clients` title?

Reply:
How about `Appservices (bridges and some bots)`?

Reply:
I went with "some" bots :)